app.js

```diff
@@ -1,35 +1,173 @@
-  if (typeof input != 'string') {
-    return input;
+var correctCount = 0; // counter for correct answers
+var questionCount = 0; // counter for number of questions asked
+
+var user = prompt('Hey, what\'s your name?');
+console.log('user name:', user);
+var hasName = user != null && user != '';
+var printUser;
+if (hasName) {
+  printUser = user;
+} else {
+  printUser = 'nameless one';
+}
+alert('Hello, ' + printUser + '. I\'m going to ask some questions about me, try and get them all right!');
+
+//YES AND NO QUESTIONS
+
+// yes/no question, correct answer, response to correct answer, response to wrong answer
+var ynQuestions = [['Do I live in Chicago?', 'N', 'Right, I actually live in Seattle.', 'Nope, I call Seattle my home.'],
+                   ['Do I have any pets?', 'Y', 'I have two cats, in fact.', 'No, well I guess my two cats own me...'],
+                   ['Am I a blackbelt in karate?', 'N', 'That is correct. No karate, just know kenjutsu.', 'I don\'t know any karate, but I do know kenjutsu.'],
+                   ['Did I write a thesis about the rearrangement of a platinocyclobutane?', 'Y', 'Yep, watched atoms moving around on a screen.', 'Wrong. And yes, that is a word.'],
+                   ['Am I Megan?', 'Y', 'Yep, that\'s my name!', 'Seriously?']];
+
+var userAns;
+var question;
+var correctAns;
+var correctResp;
+var wrongResp;
+
+for (var ask = 0; ask < ynQuestions.length; ask++) {
+  questionCount++;
+  question = ynQuestions[ask][0];
+  correctAns = ynQuestions[ask][1];
+  correctResp = ynQuestions[ask][2];
+  wrongResp = ynQuestions[ask][3];
+  do {
+    userAns = prompt(question);
+    if (typeof userAns === 'string') {
+      userAns = userAns.toUpperCase().trim();
+      userAns = userAns.charAt(0) === correctAns.charAt(0);
+    }
+    console.log(question, 'user correct:', userAns);
+  } while (typeof userAns != 'boolean')
+  if (userAns) {
+    alert(correctResp);
+    correctCount++;
   } else {
-    var tInput = input.toUpperCase().trim();
-    return tInput == 'YES' || tInput == 'Y';
+    alert(wrongResp);
   }
-};
+}
+console.log('correct y/n answers:', correctCount);
 
-/**
-Takes the question to be asked in a prompt, the answer for a response of 'yes', and the answer for a response of 'no'. Asks the question until the 'OK' button is pressed. Prints the user response as a boolean or null to the console. Responds to the user with the given answer in an alert. Returns nothing.
-**/
-var askQuestion = function(question, yesAns, noAns) {
-  var response;
-  do {
-    response = checkInput(prompt(question));
-    console.log(question, response);
-  } while (typeof response != 'boolean')
-  if (response) {
-    alert(yesAns);
+// GUESS THE NUMBER QUESTIONS
+
+// userAns defined above
+
+var favNum = Math.round((Math.random() * 100));
+console.log('my favorite number is:', favNum);
+questionCount++;
+
+var isNotDone = true;
+var message;
+
+for (var guess = 3; guess >= 0 && isNotDone; guess--) {
```

Comment:
var guess is defined two times in app.js. because it's in a for loop but not a function, it's available on the global scope